ocs_ci/deployment/deployment.py

```diff
@@ -117,6 +117,27 @@ def label_and_taint_nodes(self):
         ]
         if not worker_nodes:
             raise UnavailableResourceException("No worker node found!")
+        az_worker_nodes = {}
+        for node in worker_nodes:
+            az = node['metadata']['labels'].get(
+                'failure-domain.beta.kubernetes.io/zone'
+            )
+            az_node_list = az_worker_nodes.get(az, [])
+            az_node_list.append(node)
+            az_worker_nodes[az] = az_node_list
+        logger.info(f"Found worker nodes in AZ: {az_worker_nodes}")
+        distributed_worker_nodes = []
+        while az_worker_nodes:
+            for az in list(az_worker_nodes.keys()):
```

Comment:
you can just iterate over keys: `for key in dict:`

Reply:
Cannot, because I get RuntimeError: dictionary changed size during iteration .

But I don't want to iterate over the dictionary but just over the keys and as keys is generator is's still bounded to dictionary. So this allow me to iterate just over keys and do not touch the dictionary over the iterations when I need to really delete keys from the original dictionary.

Reply:
> Do we need convert keys into list? can we just use for az in az_worker_nodes.keys()

@vavuthu as I already explained to Vasu it raises the exception mentioned above.

If you don't trust me, here is the prove from ipython:
```python
In [1]: my_dict = {'key1': [1], 'key2': [2], 'key3': [3]}

In [2]: for key in my_dict.keys():
   ...:     if key == 'key2':
   ...:         del(my_dict[key])
   ...:
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-69e5c698d4f9> in <module>
----> 1 for key in my_dict.keys():
      2     if key == 'key2':
      3         del(my_dict[key])
      4

RuntimeError: dictionary changed size during iteration
```
or Vasu's approach:
```python
In [4]: my_dict = {'key1': [1], 'key2': [2], 'key3': [3]}

In [5]: for key in my_dict:
   ...:     if key == 'key2':
   ...:         del(my_dict[key])
   ...:
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-08cd509c1f50> in <module>
----> 1 for key in my_dict:
      2     if key == 'key2':
      3         del(my_dict[key])
      4

RuntimeError: dictionary changed size during iteration
```

and my working approach:
```python
In [6]: my_dict = {'key1': [1], 'key2': [2], 'key3': [3]}

In [7]: for key in list(my_dict.keys()):
   ...:     if key == 'key2':
   ...:         del(my_dict[key])
   ...:
   ...:

In [8]:
```

Reply:
> > Do we need convert keys into list? can we just use for az in az_worker_nodes.keys()
> 
> @vavuthu as I already explained to Vasu it raises the exception mentioned above.
> 
> If you don't trust me, here is the prove from ipython:
> 
> ```python
> In [1]: my_dict = {'key1': [1], 'key2': [2], 'key3': [3]}
> 
> In [2]: for key in my_dict.keys():
>    ...:     if key == 'key2':
>    ...:         del(my_dict[key])
>    ...:
> ---------------------------------------------------------------------------
> RuntimeError                              Traceback (most recent call last)
> <ipython-input-2-69e5c698d4f9> in <module>
> ----> 1 for key in my_dict.keys():
>       2     if key == 'key2':
>       3         del(my_dict[key])
>       4
> 
> RuntimeError: dictionary changed size during iteration
> ```
> 
> or Vasu's approach:
> 
> ```python
> In [4]: my_dict = {'key1': [1], 'key2': [2], 'key3': [3]}
> 
> In [5]: for key in my_dict:
>    ...:     if key == 'key2':
>    ...:         del(my_dict[key])
>    ...:
> ---------------------------------------------------------------------------
> RuntimeError                              Traceback (most recent call last)
> <ipython-input-5-08cd509c1f50> in <module>
> ----> 1 for key in my_dict:
>       2     if key == 'key2':
>       3         del(my_dict[key])
>       4
> 
> RuntimeError: dictionary changed size during iteration
> ```
> 
> and my working approach:
> 
> ```python
> In [6]: my_dict = {'key1': [1], 'key2': [2], 'key3': [3]}
> 
> In [7]: for key in list(my_dict.keys()):
>    ...:     if key == 'key2':
>    ...:         del(my_dict[key])
>    ...:
>    ...:
> 
> In [8]:
> ```
> 
> BTW in python 2.7 I guess your approaches will work but in python 3 it's about generators ;)

yes, it works in python2.7. got it :)